community/kernel/src/main/java/org/neo4j/kernel/impl/constraints/StandardConstraintSemantics.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.constraints;
+
+import org.neo4j.cursor.Cursor;
+import org.neo4j.kernel.api.constraints.PropertyConstraint;
+import org.neo4j.kernel.api.cursor.NodeItem;
+import org.neo4j.kernel.api.cursor.RelationshipItem;
+import org.neo4j.kernel.api.exceptions.schema.CreateConstraintFailureException;
+import org.neo4j.kernel.api.txstate.TxStateHolder;
+import org.neo4j.kernel.api.txstate.TxStateVisitor;
+import org.neo4j.kernel.impl.api.StatementOperationParts;
+import org.neo4j.kernel.impl.api.store.StoreReadLayer;
+import org.neo4j.kernel.impl.api.store.StoreStatement;
+import org.neo4j.kernel.impl.store.record.PropertyConstraintRule;
+import org.neo4j.kernel.impl.store.record.UniquePropertyConstraintRule;
+
+public class StandardConstraintSemantics implements ConstraintSemantics
+{
+    @Override
+    public void validateNodePropertyExistenceConstraint( Cursor<NodeItem> allNodes, int label, int propertyKey )
+            throws CreateConstraintFailureException
+    {
+        throw propertyExistenceConstraintsNotAllowed();
+    }
+
+    @Override
+    public void validateRelationshipPropertyExistenceConstraint( Cursor<RelationshipItem> allRels, int type,
+            int propertyKey ) throws CreateConstraintFailureException
+    {
+        throw propertyExistenceConstraintsNotAllowed();
+    }
+
+    @Override
+    public PropertyConstraint readConstraint( PropertyConstraintRule rule )
+    {
+        if ( rule instanceof UniquePropertyConstraintRule )
+        {
+            return ((UniquePropertyConstraintRule) rule).toConstraint();
+        }
+        return readNonStandardConstraint( rule );
+    }
+
+    protected PropertyConstraint readNonStandardConstraint( PropertyConstraintRule rule )
+    {
+        // todo: message and new exception type
+        throw new IllegalStateException( "Property existence constraints can only be used on Neo4j enterprise" );
```

Comment:
@akollegger this message is for when a database is loaded in Community Edition that has a property existence constraint in it (because it was created using Enterprise Edition), so `ConstraintCreationFailure` might not be an appropriate exception type here. Should the message still be the same?